File: app/src/main/java/com/example/semp/ApiService.java
package com.example.semp;

import retrofit2.http.Header;
import com.example.semp.models.*;
import java.util.List;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface ApiService {

    @POST("login")
    Call<LoginResponse> fazerLogin(@Body LoginRequest request);

    @GET("produtos")
    Call<List<Produto>> getProdutos();

    @POST("pedidos/solicitar")
    Call<GenericResponse> fazerPedido(@Body PedidoRequest request);

    @GET("pedidos")
    Call<List<PedidosPendentes>> getMeusPedidos(
            @Query("usuario") String usuario,
            @Query("nivel") String nivel,
            @Query("unidade") String unidade
    );

    @GET("pedidos/pendentes")
    Call<List<PedidosPendentes>> getPedidosPendentes();

    @GET("pedidos/todos")
    Call<List<PedidosPendentes>> getTodosPedidos(
            @Query("unidade") String unidade,
            @Query("nivel") String nivel,
            @Query("usuario") String usuario
    );

    @POST("pedidos/autorizar")
    Call<GenericResponse> autorizarPedido(@Body AutorizarRequest request);

    @POST("usuario/cadastrar")
    Call<GenericResponse> cadastrarUsuario(@Body UsuarioRequest request);

    @POST("produto/atualizar")
    Call<GenericResponse> atualizarProduto(@Body UpdateProdutoRequest request);

    @POST("produto/deletar")
    Call<GenericResponse> deletarProduto(@Body DeleteProdutoRequest request);

    @POST("produto/cadastrar")
    Call<GenericResponse> cadastrarProduto(@Body ProdutoRequest request);

    @POST("carrinho/adicionar")
    Call<GenericResponse> adicionarAoCarrinho(
            @Header("X-Usuario-ID") String usuario,
            @Body CarrinhoRequest request
    );

    @GET("carrinho")
    Call<List<Produto>> getCarrinho(
            @Header("X-Usuario-ID") String usuario
    );

    @POST("carrinho/remover")
    Call<GenericResponse> removerDoCarrinho(
            @Header("X-Usuario-ID") String usuario,
            @Body CarrinhoRequest request
    );


    // Para a tela de Rastreio
    @POST("pedido/rastreio")
    Call<GenericResponse> registrarRastreio(@Body RastreioRequest request);

    @GET("rastreio/todos")
    Call<List<Rastreio>> getTodosRastreios();

    // Para a tela de Itens Emprestados
    @GET("pedidos/emprestados")
    Call<List<PedidosPendentes>> getItensEmprestados(@Query("unidade") String unidade);

    @POST("pedidos/solicitar_retorno")
    Call<GenericResponse> solicitarRetorno(@Body RetornoRequest request);


}

File: app/src/main/java/com/example/semp/ApiService.kt
package com.example.semp

import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.Body
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.GET
import retrofit2.http.POST

data class LoginRequest(val usuario: String, val senha: String)
data class LoginResponse(val sucesso: Boolean, val mensagem: String, val usuario: String?, val nivel_conta: String?, val unidade: String?)
data class Produto(val id_estoque: Int?, val nome: String?, val codigo: String?, val descricao: String?, val quant: String?, val carrinho: Int?)
data class PedidoRequest(val nome: String, val email: String, val unidade: String, val data_reserva: String)
data class GenericResponse(val sucesso: Boolean, val mensagem: String?)

data class UsuarioRequest(val usuario: String, val senha: String, val nivel_conta: String, val unidade: String)
data class UpdateProdutoRequest(val id: Int, val coluna: String, val valor: String)
data class DeleteProdutoRequest(val codigo: String)
data class PedidoPendente(val id_emprestimo: Int, val nome: String, val unidade_natal: String, val data_reserva: String)
data class AutorizarRequest(val id_emprestimo: Int, val novoStatus: Int)

interface ApiService {
    @POST("/login") fun fazerLogin(@Body request: LoginRequest): Call<LoginResponse>
    @GET("/produtos") fun getProdutos(): Call<List<Produto>>
    @GET("/carrinho") fun getCarrinho(): Call<List<Produto>>
    @POST("/pedido/fazer") fun fazerPedido(@Body request: PedidoRequest): Call<GenericResponse>

    @POST("/usuario/cadastrar") fun cadastrarUsuario(@Body request: UsuarioRequest): Call<GenericResponse>
    @POST("/produto/atualizar") fun atualizarProduto(@Body request: UpdateProdutoRequest): Call<GenericResponse>
    @POST("/produto/deletar") fun deletarProduto(@Body request: DeleteProdutoRequest): Call<GenericResponse>

    @GET("/pedidos/pendentes") fun getPedidosPendentes(): Call<List<PedidoPendente>>
    @POST("/pedidos/autorizar") fun autorizarPedido(@Body request: AutorizarRequest): Call<GenericResponse>

    // CORREÇÃO: Transforma o envio em Form-Data nativo do PHP!
    @FormUrlEncoded
    @POST("adicionar_carrinho.php")
    fun adicionarAoCarrinho(
        @Field("id_produto") id_produto: String,
        @Field("quantidade") quantidade: Int
    ): Call<GenericResponse>
}

object RetrofitClient {
    private const val BASE_URL = "https://api-estoque.whyguiih.workers.dev/"
    val api: ApiService by lazy { Retrofit.Builder().baseUrl(BASE_URL).addConverterFactory(GsonConverterFactory.create()).build().create(ApiService::class.java) }
}